File: src/main/java/com/nir/beans/GeneralizedMethodInfoComponentImpl.java
package com.nir.beans;

import com.nir.utils.math.method.generalized.GeneralizedMethodInfoJsonPojo;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

public class GeneralizedMethodInfoComponentImpl implements GeneralizedMethodInfoComponent {

    private List<JacksonComponent<GeneralizedMethodInfoJsonPojo>> jacksonComponents;

    public GeneralizedMethodInfoComponentImpl(List<JacksonComponent<GeneralizedMethodInfoJsonPojo>> jacksonComponents) {
        this.jacksonComponents = jacksonComponents;
    }

    public List<GeneralizedMethodInfoJsonPojo> getAll() {
        return jacksonComponents.stream()
            .map(JacksonComponent::readAll).flatMap(Collection::stream)
            .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/nir/beans/method/GeneralizedMethodInfoComponentImpl.kt
package com.nir.beans.method

import com.nir.beans.method.generalized.GeneralizedMethodInfoJsonPojo
import com.nir.beans.reaction.parsing.JacksonComponent
import java.util.stream.Collectors

class GeneralizedMethodInfoComponentImpl(private val jacksonComponents: List<JacksonComponent<out GeneralizedMethodInfoJsonPojo>>) :
    GeneralizedMethodInfoComponent {
    override val getMethods: List<GeneralizedMethodInfoJsonPojo>
        get() = jacksonComponents.stream()
                .map { obj: JacksonComponent<out GeneralizedMethodInfoJsonPojo> -> obj.readAll() }.flatMap { obj: List<GeneralizedMethodInfoJsonPojo> -> obj.stream() }
                .collect(Collectors.toList())
}